lib/semian/circuit_breaker.rb

```diff
@@ -43,70 +44,77 @@ def mark_failed(_error)
 
     def mark_success
       return unless half_open?
-      @successes += 1
+      @successes.increment
       close if success_threshold_reached?
     end
 
     def reset
-      @errors = []
-      @successes = 0
+      @errors.clear
+      @successes.value = 0
```

Comment:
I'd add a reset / clear method to Successes too. If we want to expose an API sutitable for atomic operations, we shouldn't encourage to set arbitrary state, just `increment` and `reset`.